src/InputParser.jl

```diff
@@ -0,0 +1,379 @@
+module InputParser
+
+# Julia implicitly numbers Enum items 0,1,2,3,4 (in order)
+# I put them in same order as FORTRAN code (i.e. NOPT=0 meant
+# ConsolidationSwell so Int(ConsolidationSwell) = 0) however
+# this wont matter since we should never be referring to the integer 
+# values explicitly anyways
+@enum Model ConsolidationSwell LeonardFrost Schmertmann CollapsibleSoil SchmertmannElastic 
+@enum Foundation ErrorFoundation RectangularSlab LongStripFooting 
+
+# These are variables/objects visible to all modules
+# that include InputParser module
+export Model, Foundation, InputData
+
+# This code repeats many times in InputData constructor
+# Parses String at input[index] and returns an array A such that size(A) == items
+function parseCurrentLine(input::Array{String}, items::Int, index::Int)
+    currentLine = input[index]
+    # Tabs ruin splitting by space
+    currentLine = replace(currentLine, "\t" => " ")
+    currentLineData = split(currentLine, " ")
+    currentLineData = filter(x -> x != "", currentLineData)
+    # Check if data is there
+    if size(currentLineData)[1] < items
+        println("Error: Invalid input file!")
+        println("\t>Line $(index): $(items) values expected")
+        return -1
```

Comment:
Would it be better to raise an Exception here, rather than use a convention that returning -1 means something went wrong?

Reply:
Yes I plan on learning proper error handling / exception throwing soon. Part of the TODO comments. Would you like that as the step?

Reply:
No, you don't have to add exception handling right now.  Rather than just relying on the TODO comment though, please create an issue.

Reply:
new issue added: #41